src/Features/Core/Portable/CodeLens/CodeLensReferencesService.cs

```diff
@@ -180,7 +179,13 @@ private static SyntaxNode GetEnclosingCodeElementNode(Document document, SyntaxT
 
                     await Task.WhenAll(referenceTasks).ConfigureAwait(false);
 
-                    return referenceTasks.Select(task => task.Result);
+                    var result = ArrayBuilder<ReferenceLocationDescriptor>.GetInstance();
```

Comment:
Since we know the length of `referenceTasks` should this be initialized with a capacity?